### PR TITLE
[FIX] account: apply base tag to cash basis entries properly

### PR DESCRIPTION
**Pull request by oco-odoo**
When performing reconciliation and creating the cash basis entry for a tax based on payment, the base move lines never received any tag, which was plain wrong and made the tax reports inconsistent.

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -1762,14 +1762,22 @@ def _get_tax_cash_basis_base_key(self, tax, move, line):
...
 
     def _get_tax_cash_basis_base_common_vals(self, key, new_move):
-        line, account_id, tax, currency_id, partner_id = key
-        line = self.env['account.move.line'].browse(line)
+        line_id, account_id, tax_id, currency_id, partner_id = key
+
+        line = self.env['account.move.line'].browse(line_id)
+        tax = self.env['account.tax'].browse(tax_id)
+
+        orig_inv_type = (self.debit_move_id + self.credit_move_id).mapped('invoice_id.type')
```
Comment (smetl) [inline addons/account/models/account_move.py:1770]: are you sure this method is `ensure_one`?

File: addons/account/tests/test_reconciliation.py [hunk 2]
```diff
@@ -2117,6 +2135,23 @@ def test_reconciliation_cash_basis_foreign_currency_low_values(self):
...
+
+        caba_base_line = caba_move.line_ids.filtered(lambda x: x.tax_ids)
+        caba_tax_line = caba_move.line_ids.filtered(lambda x: x.tax_line_id)
+        other_lines = caba_move.line_ids - (caba_base_line + caba_tax_line)
+
+        self.assertEqual(len(caba_base_line), 1, "Exactly one line should exist with tax_ids set on the cash basis move")
+        self.assertEqual(len(caba_tax_line), 1, "Exactly one line should exist with tax_line_id set on the cash basis move")
+
+        self.assertEqual(caba_base_line.tag_ids, self.tax_tag_base, "The base line of the cash basis move should have received the base tag")
+        self.assertEqual(caba_tax_line.tag_ids, self.tax_tag_tax, "The tax line of the cash basis move should have received the tax tag")
```
Comment (smetl) [inline addons/account/tests/test_reconciliation.py:2152]:
```
self.assertRecordValues(caba_base_line + caba_tax_line, [
  {'tag_ids': self.tax_tag_base.ids},
  {'tag_ids': self.tax_tag_tax.ids},
])
```
?

File: addons/account/models/account_move.py (hunk 1 above)
Comment (oco-odoo) [reply addons/account/models/account_move.py:1770]: Yup, its only call is triggered by create_tax_cash_basis_entry (through an intermediate function called only once as well), which is ensure_one. I'll still add the check, though.

File: addons/account/tests/test_reconciliation.py (hunk 2 above)
Comment (oco-odoo) [reply addons/account/tests/test_reconciliation.py:2152]: Alright, I won't say "Ni" this time :p

**Comment by smetl** [comment]
@oco-odoo It seems a tax report test is broken with this fix :x

**Comment by oco-odoo** [comment]
@smetl What ? :o How come the branch is green ? :o

**Comment by smetl** [comment]
@oco-odoo The community branch is green but not the enterprise one

**Comment by oco-odoo** [comment]
@smetl There is no enterprise branch :p (I mean, no dev one) This test is the one I introduced in another branch yesterday, probably that's why it crashed only when robodoo was trying to merge ^^ Anyway, I'm having a look right now.

**Comment by oco-odoo** [comment]
@smetl Should be ok now :)

**Comment by oco-odoo** [comment]
@smetl Un p'tit r+ when you're back ? :)

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**